src/controllers/postController.ts

```diff
@@ -0,0 +1,69 @@
+// postController.ts
+import { Request, RequestHandler, Response } from "express";
+import { StatusCodes } from "http-status-codes";
+import { PostError } from "../errors/customErrors.ts";
+import { Post } from "../models/Post.ts";
+import {
+  buildPostDocument,
+  deletePostDocument,
+  updatePostDocument,
+} from "../services/postServices.ts";
+
+const createPostController = async (req: Request) => {
+  const postBody = req.body;
+  const createdPost = await buildPostDocument(postBody);
+  if (!createdPost) {
+    return null;
+  }
+  return { message: "Post created", data: createdPost };
+};
+
+const updatePostController = async (req: Request) => {
+  const postBody = req.body;
+  const { postid } = req.params;
+  const updatedPost = await updatePostDocument(postid, postBody);
+  if (!updatedPost) {
+    return null;
+  }
+  return {
+    message: "Post updated",
+    data: updatedPost,
+  };
+};
+
+const deletePostController = async (req: Request) => {
+  const { postid } = req.params;
+  await deletePostDocument(postid);
+  return { message: "Post deleted" };
+};
+
+export const postService = async (req: Request, res: Response) => {
+  try {
+    let response;
+    switch (req.method) {
```

Comment:
you should not create a post services. Import the corresponding function directly into the post Router and put it there.